### PR TITLE
nf-test fgbio/fastqtobam + small updates

### PR DESCRIPTION
**Pull request by nvnieuwk**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by pinin4fjords — APPROVED** [review]
LGTM

File: modules/nf-core/fgbio/fastqtobam/tests/bam.config [hunk 1]
```diff
@@ -0,0 +1,3 @@
+process {
+    ext.suffix = "bam"
+}
```
Comment (mashehu) [inline modules/nf-core/fgbio/fastqtobam/tests/bam.config:3]: is this needed? it's the default value anyway

File: modules/nf-core/fgbio/fastqtobam/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.bam).match("bam")},
+                { assert snapshot(process.out.cram).match("cram")},
+                { assert snapshot(process.out.versions).match("versions") }
```
Comment (mashehu) [inline modules/nf-core/fgbio/fastqtobam/tests/main.nf.test:33]:
```suggestion
                { assert snapshot(process.out).match()}
```
Why not like this?

File: tests/config/pytest_modules.yml [hunk 3]
```diff
@@ -1120,10 +1120,6 @@ fgbio/callmolecularconsensusreads:
   - modules/nf-core/fgbio/callmolecularconsensusreads/**
   - tests/modules/nf-core/fgbio/callmolecularconsensusreads/**
 
-fgbio/fastqtobam:
-  - modules/nf-core/fgbio/fastqtobam/**
-  - tests/modules/nf-core/fgbio/fastqtobam/**
-
```
Comment (mashehu) [inline tests/config/pytest_modules.yml:1126]: please remove also the old pytest files themselves.

**Review by mashehu — COMMENTED** [review]
please remove the old pytest a

File: modules/nf-core/fgbio/fastqtobam/tests/bam.config (hunk 1 above)
Comment (nvnieuwk) [reply modules/nf-core/fgbio/fastqtobam/tests/bam.config:3]: Not really, but I took this from the old tests where they tested it too. Do you want me to remove it?

File: modules/nf-core/fgbio/fastqtobam/tests/main.nf.test (hunk 2 above)
Comment (nvnieuwk) [reply modules/nf-core/fgbio/fastqtobam/tests/main.nf.test:33]: Good point, although the md5sums are different I noticed so I'll have some fixes coming up

File: tests/config/pytest_modules.yml (hunk 3 above)
Comment (nvnieuwk) [reply tests/config/pytest_modules.yml:1126]: Done!

**Pull request merged**